[CI] MessagesTests testJoinRequestEqualsHashCodeSerialization failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/ykjizrd5cc2hw/tests/:server:test/org.elasticsearch.cluster.coordination.MessagesTests/testJoinRequestEqualsHashCodeSerialization
**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.MessagesTests.testJoinRequestEqualsHashCodeSerialization" -Dtests.seed=A18E16EB6247A000 -Dtests.locale=sr-Latn-ME -Dtests.timezone=Etc/GMT-10 -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.MessagesTests&tests.test=testJoinRequestEqualsHashCodeSerialization


**Failure excerpt:**
```
java.lang.AssertionError: JoinRequest mutation should not be equal to original
Expected: not <JoinRequest{sourceNode={CyxhPtFNkh}{HiHo4cPKQmu5i3kegM6Ogw}{0.0.0.0}{0.0.0.0:9}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}, compatibilityVersions=CompatibilityVersions[transportVersion=8020099, systemIndexMappingsVersion={.hxiPW=MappingsVersion[version=14, hash=1133181213], .pbSdI=MappingsVersion[version=14, hash=7798135], .VepwA=MappingsVersion[version=0, hash=-718584528]}], features=[], minimumTerm=2437835834329717953, optionalJoin=Optional[Join{term=2572156991087473853, lastAcceptedTerm=8520950879130564435, lastAcceptedVersion=7761831410985385488, sourceNode={CyxhPtFNkh}{HiHo4cPKQmu5i3kegM6Ogw}{0.0.0.0}{0.0.0.0:9}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}, targetNode={vvMwhUlwyX}{OYtEkRRqRQWKn32g6nnupQ}{0.0.0.0}{0.0.0.0:10}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}}]}>
     but: was <JoinRequest{sourceNode={CyxhPtFNkh}{HiHo4cPKQmu5i3kegM6Ogw}{0.0.0.0}{0.0.0.0:9}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}, compatibilityVersions=CompatibilityVersions[transportVersion=8020099, systemIndexMappingsVersion={.hxiPW=MappingsVersion[version=14, hash=1133181213], .pbSdI=MappingsVersion[version=14, hash=7798135], .VepwA=MappingsVersion[version=0, hash=-718584528]}], features=[], minimumTerm=2437835834329717953, optionalJoin=Optional[Join{term=2572156991087473853, lastAcceptedTerm=8520950879130564435, lastAcceptedVersion=7761831410985385488, sourceNode={CyxhPtFNkh}{HiHo4cPKQmu5i3kegM6Ogw}{0.0.0.0}{0.0.0.0:9}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}, targetNode={vvMwhUlwyX}{OYtEkRRqRQWKn32g6nnupQ}{0.0.0.0}{0.0.0.0:10}{IScdfhilmrstvw}{8.12.0}{7000099-8500003}}]}>

  at __randomizedtesting.SeedInfo.seed([A18E16EB6247A000:7C083AA4AB09BDC3]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:95)
  at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:63)
  at org.elasticsearch.cluster.coordination.MessagesTests.testJoinRequestEqualsHashCodeSerialization(MessagesTests.java:233)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/MessagesTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/MessagesTests.java
@@ -215,7 +215,6 @@ public class MessagesTests extends ESTestCase {
         );
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/101733")
     public void testJoinRequestEqualsHashCodeSerialization() {
         Join initialJoin = new Join(
             createNode(randomAlphaOfLength(10)),
@@ -258,7 +257,7 @@ public class MessagesTests extends ESTestCase {
                 case 2 -> new JoinRequest(
                     joinRequest.getSourceNode(),
                     joinRequest.getCompatibilityVersions(),
-                    Set.of(generateRandomStringArray(10, 10, false)),
+                    randomValueOtherThan(joinRequest.getFeatures(), () -> Set.of(generateRandomStringArray(10, 10, false))),
                     joinRequest.getMinimumTerm(),
                     joinRequest.getOptionalJoin()
                 );

